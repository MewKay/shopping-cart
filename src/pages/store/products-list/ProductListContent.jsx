import PropTypes from "prop-types";
import { AlertTriangle } from "lucide-react";
import ProductCard from "./ProductCard";
import styles from "./ProductListContent.module.css";

const ProductListContent = ({
  loading,
  error,
  isThereNoProducts,
  filteredProductList,
  onCardClick,
}) => {
  return (
    <div className={styles["list-content"]}>
      {loading ? (
        <p>Loading...</p>
      ) : error ? (
        <div className={styles["alert-message"]}>
          <AlertTriangle />
          <p>Failed to load products. Please try again later.</p>
        </div>
      ) : isThereNoProducts ? (
        <div className={styles["alert-message"]}>
          <AlertTriangle />
          <p>
            Sorry, no products match your search. Please try adjusting your
            filters or search terms.
          </p>
        </div>
      ) : (
        <div className={styles["products-list"]}>
          {filteredProductList.map((product, index) => (
            <ProductCard
              key={product.id}
              imgURL={product.image}
              title={product.title}
              price={product.price}
              indexToShow={index}
              onClick={onCardClick}
            />
          ))}
        </div>
      )}
    </div>
  );
};

ProductListContent.propTypes = {
  loading: PropTypes.bool.isRequired,
  error: PropTypes.any,
  isThereNoProducts: PropTypes.bool.isRequired,
  filteredProductList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      image: PropTypes.string,
      title: PropTypes.string,
      price: PropTypes.number,
    })
  ),
  onCardClick: PropTypes.func.isRequired,
};

export default ProductListContent;
