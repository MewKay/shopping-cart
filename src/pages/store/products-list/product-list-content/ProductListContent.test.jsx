import { describe, it, expect, vi } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ProductListContent from "./ProductListContent";

//Components props
const loading = false;
const error = null;
const isThereNoProducts = false;
const filteredProductList = [
  {
    id: 0,
    image: "https://image.placeholder.com/product",
    title: "Product 0",
    price: 49.5,
  },
  {
    id: 1,
    image: "https://image.placeholder.com/product",
    title: "Product 1",
    price: 22.5,
  },
  {
    id: 2,
    image: "https://image.placeholder.com/product",
    title: "Product 2",
    price: 10.99,
  },
];
const onCardClick = vi.fn();

describe("ProductListContent", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ProductListContent
        loading={loading}
        error={error}
        isThereNoProducts={isThereNoProducts}
        filteredProductList={filteredProductList}
        onCardClick={onCardClick}
      />,
      {
        wrapper: MemoryRouter,
      }
    );

    expect(container).toMatchSnapshot();
  });
});
