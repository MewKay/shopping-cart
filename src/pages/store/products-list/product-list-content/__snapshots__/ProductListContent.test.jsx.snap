// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductListContent > renders correctly 1`] = `
<div>
  <div
    class="_list-content_6eac38"
  >
    <div
      class="_products-list_6eac38"
    >
      <button
        class="_container_454344"
      >
        <div
          class="_product-image_454344"
        >
          <img
            alt=""
            src="https://image.placeholder.com/product"
          />
        </div>
        <p
          class="_item-title_454344"
        >
          Product 0
        </p>
        <p
          class="_item-price_454344"
        >
          $
          49.5
        </p>
      </button>
      <button
        class="_container_454344"
      >
        <div
          class="_product-image_454344"
        >
          <img
            alt=""
            src="https://image.placeholder.com/product"
          />
        </div>
        <p
          class="_item-title_454344"
        >
          Product 1
        </p>
        <p
          class="_item-price_454344"
        >
          $
          22.5
        </p>
      </button>
      <button
        class="_container_454344"
      >
        <div
          class="_product-image_454344"
        >
          <img
            alt=""
            src="https://image.placeholder.com/product"
          />
        </div>
        <p
          class="_item-title_454344"
        >
          Product 2
        </p>
        <p
          class="_item-price_454344"
        >
          $
          10.99
        </p>
      </button>
    </div>
  </div>
</div>
`;
